### ISSUE TITLE
Fix the width of completion menu

### ISSUE THREAD

**Issue by krishnakumarg1984** [hrsh7th/nvim-cmp#980]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/15f08a8faa22d52480cdcb9ef9ca698120f04363/doc/cmp.txt#L616) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/hrsh7th/nvim-cmp/issues) and there are no open or closed issues with the same problem.

### Neovim Version

0.7

### Minimal reproducible full config

`cmp.nvim`'s autocompletion pop-up menu changes width to accommodate the completion candidates.

While, there are a few options to set the max_width, I don't see any solution to fix the pop-up menu at a certain width.
Can we have an option in `cmp` to set the pop-up menu to a fixed width? 

### Description

It is rather annoying when the completion menu's width keeps on changing to be bigger and smaller, leading to a very jarring user experience.

### Steps to reproduce

Default config of `cmp`. 

### Expected behavior

Fix the pop-up window width irrespective of the length of the completion candidates.

### Actual behavior

The completion menu's width keeps on changing to be bigger and smaller dynamically, depending on the width of the populated candidates. 

### Additional context

The constantly changing pop-up menu width is rather annoying and leads to a jarring user experience.

**Comment by gegoune** [comment]
https://github.com/hrsh7th/nvim-cmp/discussions/609#discussioncomment-1844480 Is that solution good enough?

**Comment by krishnakumarg1984** [comment]
No. As I mentioned in my comments to that question. The solution in [#609 (comment)](https://github.com/hrsh7th/nvim-cmp/discussions/609#discussioncomment-1844480) only helps to truncate items to a maximum width.

It doesn't help to eliminate the jarring visual effects of changing pop-up width during completion.

**Comment by Shougo** [comment]
> No. As I mentioned in my comments to that question. The solution in https://github.com/hrsh7th/nvim-cmp/discussions/609#discussioncomment-1844480 only helps to truncate items to a maximum width.

I think you can define minimum width in `format` function.

**Comment by krishnakumarg1984** [comment]
@Shougo Nice. Can you show us how?

**Comment by Shougo** [comment]
```lua
  local ELLIPSIS_CHAR = '…'
  local MAX_LABEL_WIDTH = 20
  local MIN_LABEL_WIDTH = 20

  cmp.setup({
    formatting = {
      format = function(entry, vim_item)
        local label = vim_item.abbr
        local truncated_label = vim.fn.strcharpart(label, 0, MAX_LABEL_WIDTH)
        if truncated_label ~= label then
          vim_item.abbr = truncated_label .. ELLIPSIS_CHAR
        elseif string.len(label) < MIN_LABEL_WIDTH then
          local padding = string.rep(' ', MIN_LABEL_WIDTH - string.len(label))
          vim_item.abbr = label .. padding
        end
        return vim_item
      end,
    },
  })
```

**Comment by krishnakumarg1984** [comment]
@Shougo Awesome. Thank works! Thank you so much!

**Comment by kevinhughes27** [comment]
It looks like the menu still saves space for the menu/kind even when `fields` is set to only `abbr`. This solution only fixes the width of the `abbr` field as far as I can tell when I tried it. I'm using only the abbr field for cmdline completion where the other fields are not useful. It would be great to remove this extra space!

**Comment by askfiy** [comment]
Wish he had built in

**Comment by krishnakumarg1984** [comment]
@hrsh7th Can we please have a `cmp` menu with overall fixed width, as well as a fixed width for the description field?

**Comment by nullptr-yxw** [comment]
> Wish he had built in

As was mentioned in #1535, It might be possible to set the minimum width and maximum height by options `pumwidth` and `pumheight`, but they are global settings. So I will push a new issue about it

**Comment by gmr458** [comment]
> It looks like the menu still saves space for the menu/kind even when `fields` is set to only `abbr`. This solution only fixes the width of the `abbr` field as far as I can tell when I tried it. I'm using only the abbr field for cmdline completion where the other fields are not useful. It would be great to remove this extra space!
> 
> Update - I can achieve what I want by setting kind and menu to empty string:
> 
> ```lua
> format = function(_, vim_item)
>     vim_item.menu = ""
>     vim_item.kind = ""
>     return vim_item
> end
> ```

this is working for me, I only use `kind` and `abbr` so I set `vim_item.menu = ''`
